charts/incubator/privatebin/templates/_secrets.tpl

```diff
@@ -0,0 +1,225 @@
+{{/*
+  Custom PrivateBin configuration. See also:
+  https://github.com/PrivateBin/docker-nginx-fpm-alpine#custom-configuration
+
+  The default configuration file can be found here:
+  https://github.com/PrivateBin/PrivateBin/blob/master/cfg/conf.sample.php
+*/}}
+{{- define "privatebin.secrets" -}}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: privatebin-secret
+  labels:
```

Comment:
I think all this file would be better fit for configmap?

Reply:
Also this file will be read-only and won't let user edit it.

If there are any options that would be frequently need to be changed, you might have to make them configurable
for example basepath

Reply:
> I think all this file would be better fit for configmap?

It contains sensitive Postgres connection info, so my thinking was to use a secret instead. Is there any downside to using a secret?

> If there are any options that would be frequently need to be changed, you might have to make them configurable
for example basepath

I'll add an option to configure the `basepath`. PrivateBin also allows changing PHP / nginx parameters by enqueuing additional files in the respective `conf.d` dirs. Can I do that in an additional PR before moving it to `stable`?

Also, I forgot to mount the config file 😂

Reply:
I went ahead and exposed all the `conf.php` options to values.yaml / questions.yaml. It feels weird having to duplicate default values in both values.yaml and questions.yaml.

I'm not sure if this was the correct approach but I required a couple of "nullable booleans". The way I did this was by declaring `string` type variables and added `true` and `false` to the `enum` field.

Reply:
> I went ahead and exposed all the `conf.php` options to values.yaml / questions.yaml. It feels clunky having to duplicate default values in both values.yaml and questions.yaml.

I agree, I have a big project in my head (and my todo list) to autogenerate the whole damn questions.yaml based on values.yaml. But sadly we can't do without `questions.yaml`. But SCALE UI needs to know what to render, and then pass changes to values.yaml.


The only way I see this changing is that they drop the UI and give access to values.yaml. But I don't think this is going to happen anytime soon

> I'm not sure if this was the correct approach but I required a couple of "nullable booleans". The way I did this was by declaring `string` type variables and added `true` and `false` to the `enum` field.

Yep, done it myself couple of times.

Reply:
> I agree, I have a big project in my head (and my todo list) to autogenerate the whole damn questions.yaml based on values.yaml.

That would be amazing. I thought about how nice it would be to have a WYSIWYG side by side editor / validator.

Especially getting the "sections" right was pretty finicky (what's removable and what's required) for someone just starting out like me:

> # Include{persistenceBasic}

This isn't helm syntax right? Is this proprietary TrueNAS Web GUI syntax and documented somewhere? Is it possible to unit test the semantics of questions.yaml?

Reply:
This is just a placeholder, where a shell script replaces it before app gets pushed to `truecharts/catalog`
Here is the actual code behind it.
https://github.com/truecharts/apps/tree/master/templates/questions

Looking at those, you would realize that the questions.yaml is already a pretty slimmed down version of what it ends up looking. 
(Check a questions.yaml in catalog repo. Its huge!)

Reply:
There is a validator running for questions on the CI, but it's just code copied from iX. We don't have a clear schema.
There is some docs for it here https://github.com/truenas/charts, but from my experience there are things that missing or not clearly explained.